modules/002-deckhouse/hooks/check_deckhouse_release.go

```diff
@@ -484,6 +503,76 @@ func (dcr *DeckhouseReleaseChecker) CalculateReleaseDelay(ts time.Time, clusterU
 	return nil
 }
 
+func (dcr *DeckhouseReleaseChecker) StepByStepUpdate(actual, target *semver.Version) error {
+	nextVersion, err := dcr.nextVersion(actual, target)
+	if err != nil {
+		return err
+	}
+	if nextVersion == target {
+		return nil
+	}
+
+	image, err := dcr.registryClient.Image(nextVersion.Original())
+	if err != nil {
+		return err
+	}
+
+	releaseMeta, err := dcr.fetchReleaseMetadata(image)
+	if err != nil {
+		return err
+	}
+	if releaseMeta.Version == "" {
+		return fmt.Errorf("version not found. Probably image is broken or layer does not exist")
+	}
+
+	dcr.releaseMetadata = releaseMeta
+
+	return nil
+}
+
+func (dcr *DeckhouseReleaseChecker) nextVersion(actual, target *semver.Version) (*semver.Version, error) {
+	if actual.Major() != target.Major() {
+		return nil, fmt.Errorf("major version updated") // TODO step by step update for major version
+	}
+
+	if actual.Minor() == target.Minor() || actual.IncMinor().Minor() == target.Minor() {
+		return target, nil
+	}
+
+	listTags, err := dcr.registryClient.ListTags()
+	if err != nil {
+		return nil, err
+	}
+
+	minor := strconv.FormatInt(int64(actual.IncMinor().Minor()), 10)
+	expr := fmt.Sprintf("^v1.%s.([0-9]+)$", minor)
+	r, err := regexp.Compile(expr)
+	if err != nil {
+		dcr.logger.Error(err)
+		return nil, err
+	}
+
+	collection := make([]*semver.Version, 0)
+	for _, ver := range listTags {
+		if r.MatchString(ver) {
+			newSemver, err := semver.NewVersion(ver)
+			if err != nil {
+				dcr.logger.Warn(err)
```

Comment:
we really need logging here? if yes, maybe add more explicit error? not only "can't parse semver"

Reply:
we need logging, because here we don't stop at the error. We skip the invalid version and move on.

Reply:
so, maybe add more verbosity to error?